### PR TITLE
no-store_v2:  Store ConfState as part of WAL log snapshot

### PR DESCRIPTION
**Pull request by ptabor**
WAL log snapshots used to contain only Term & Index. 
That made them diverging from raft.Snapshot metadata that contained ConfState as well
(list of Voters, Listeners, and candidating voters & listeners). 
So far lookup of the full ConfState required dive into *.snap file that is reminiscence of store_v2. 

I'm working on preparing for decomissioning of store_v2 in 3.6, so in particular etcd-3.5 
should be able to run in situation that *.snap files does not exist (to be rollback ready).

### CODE REVIEW

File: server/wal/wal_test.go [hunk 1]
```diff
@@ -35,8 +35,15 @@ import (
...
+var (
+	confState = raftpb.ConfState{
+		Voters:    []uint64{0x00ffca74},
+		AutoLeave: false,
+	}
+)
+
 func TestNew(t *testing.T) {
-	p, err := ioutil.TempDir(os.TempDir(), "waltest")
+	p, err := ioutil.TempDir(t.TempDir(), "waltest")
```
Comment (gyuho) [inline server/wal/wal_test.go:46]: nice :)

File: server/wal/wal.go [hunk 2]
```diff
@@ -942,6 +941,10 @@ func (w *WAL) Save(st raftpb.HardState, ents []raftpb.Entry) error {
 }
 
 func (w *WAL) SaveSnapshot(e walpb.Snapshot) error {
+	if err := walpb.ValidateSnapshotForWrite(&e); err != nil {
```
Comment (gyuho) [inline server/wal/wal.go:944]: Let's say a, b, c

And c as a follower runs this code against a as a leader with old etcd version. Would the follower reject the snapshot from old-versioned leader?

Comment (ptabor) [reply server/wal/wal.go:944]: I think there would be no problem as raft is speaking using `raftpb.Snapshot`

That always contains 
https://github.com/etcd-io/etcd/blob/102c1984444c0bc971f5bd0be49423840f6d4351/raft/raftpb/raft.proto#L33
with: 
https://github.com/etcd-io/etcd/blob/102c1984444c0bc971f5bd0be49423840f6d4351/raft/raftpb/raft.proto#L26. 

This validation just makes sure we never forgot to copy the content from raftpb.Snapshot to walpb.Snapshot. 
That cannonically happens in server/etcdserver/storage.go (line 59)

**Comment by ptabor** [comment]
@gyuho Shell I merge, or your concern remains ?

File: server/wal/wal.go (hunk 2 above)
Comment (gyuho) [reply server/wal/wal.go:946]: @ptabor +1. Yeah, this is for saving the snap locally.

**Pull request merged**